packages/wrangler/src/type-generation.ts

```diff
@@ -112,6 +112,47 @@ export async function typesHandler(
 	);
 }
 
+/**
+ * Check if a string is a valid TypeScript identifier. This is a naive check and doesn't cover all cases
+ * If not, it should be wrapped in quotes
+ */
+export function isValidIdentifier(key: string) {
+	return /^[a-zA-Z_$][\w$]*$/.test(key);
+}
+
+/**
+ * Construct a type key, if it's not a valid identifier, wrap it in quotes
+ */
+export function constructTypeKey(key: string) {
+	if (isValidIdentifier(key)) {
+		return `${key}`;
+	}
+	return `"${key}"`;
+}
+
+/**
+ *
+ * Construct a type key, if it's not a valid identifier, wrap it in quotes
+ * If rawVal is true, we'll use the value as the type, otherwise we'll format it for a string definition
+ */
+export function constructType(
+	key: string,
+	value: string | number | boolean,
+	rawVal = true
+) {
+	const typeKey = constructTypeKey(key);
+	if (typeof value === "string") {
+		if (rawVal) {
+			return `${typeKey}: ${value};`;
+		}
+		return `${typeKey}: "${value.replace(/"/g, '\\"')}";`;
+	}
+	if (typeof value === "number" || typeof value === "boolean") {
+		return `${typeKey}: ${value};`;
+	}
+	return `${typeKey}: unknown;`;
```

Comment:
I _think_ this function could be simplified:
```suggestion
	const typeKey = constructTypeKey(key);

	if(rawVal && typeof value === 'string') {
		return `${typeKey}: ${value};`;
	}

	return `${typeKey}: "${`${value}`.replace(/"/g, '\\"')}";`;
```

Reply:
If I'm understanding correctly, that's very different logically, and doesn't handle numbers/booleans like the previous implementation did.

`constructType('valid', 2)`:

Previous: `'valid: 2;'`
New: `'valid: "2";'`

Reply:
yes, you're right! sorry I must have missed something in the code 🙇 

it looks good to me 🙂 

but you could please then add other unit tests for the non-string scenarios too? 🙏 

(such as `expect(constructType("valid", 2)).toBe("valid: 2;");` for example 🙂 )